pkg/liqo-controller-manager/core/foreigncluster-controller/status.go

```diff
@@ -126,15 +131,50 @@ func (r *ForeignClusterReconciler) handleNetworkingModuleStatus(ctx context.Cont
 		}
 	}
 
+	connection, err := getters.GetConnectionByClusterID(ctx, r.Client, string(clusterID))
+	switch {
+	case errors.IsNotFound(err):
+		klog.V(6).Infof("Connection resource not found for ForeignCluster %q", clusterID)
+		fcutils.DeleteModuleCondition(&fc.Status.Modules.Networking, liqov1beta1.NetworkConnectionStatusCondition)
+		statusExceptions[liqov1beta1.NetworkConnectionStatusCondition] = StatusException{
+			ConditionStatusType: liqov1beta1.ConditionStatusNotReady,
+			Reason:              connectionMissingReason,
+			Message:             connectionMissingMessage,
+		}
+	case err != nil:
+		klog.Errorf("an error occurred while getting the Connection resource for the ForeignCluster %q: %s", clusterID, err)
+		return err
+	default:
+		fcutils.EnableModuleNetworking(fc)
+		switch connection.Status.Value {
+		case networkingv1beta1.Connected:
+			fcutils.EnsureModuleCondition(&fc.Status.Modules.Networking,
+				liqov1beta1.NetworkConnectionStatusCondition, liqov1beta1.ConditionStatusEstablished,
+				connectionEstablishedReason, connectionEstablishedMessage)
+		case networkingv1beta1.Connecting:
+			fcutils.EnsureModuleCondition(&fc.Status.Modules.Networking,
+				liqov1beta1.NetworkConnectionStatusCondition, liqov1beta1.ConditionStatusPending,
+				connectionPendingReason, connectionPendingMessage)
+		default:
+			fcutils.EnsureModuleCondition(&fc.Status.Modules.Networking,
+				liqov1beta1.NetworkConnectionStatusCondition, liqov1beta1.ConditionStatusError,
+				connectionErrorReason, connectionErrorMessage)
+		}
+	}
+
 	gwClient, err := getters.GetGatewayClientByClusterID(ctx, r.Client, clusterID)
 	switch {
 	case errors.IsNotFound(err):
 		klog.V(6).Infof("GatewayClient resource not found for ForeignCluster %q", clusterID)
 		fcutils.DeleteModuleCondition(&fc.Status.Modules.Networking, liqov1beta1.NetworkGatewayClientStatusCondition)
+		// Do not write any exception, as if both gateways are missing we write a single condition telling that a gateway is missing.
+		// If instead, only the gateway client is missing, then this cluster acts as server.
```

Comment:
I would suggest to simplify the workflow. You can get the gateway client and server and add the condition only if the 2 arrors are NotFound